src/mlpack/core/data/save_image.hpp

```diff
@@ -9,39 +9,34 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
+#ifndef MLPACK_CORE_DATA_SAVE_IMAGE_HPP
+#define MLPACK_CORE_DATA_SAVE_IMAGE_HPP
+
 #include "save.hpp"
 
 #ifdef HAS_STB
 
-// The implementation of the functions is included directly, so we need to make
-// sure it doesn't get included twice.  This is to work around a bug in old
-// versions of STB where not all functions were correctly marked static.
+// Include STB functions.  Note that we include the implementations, too, and
+// all functions will be marked as static.
 #define STB_IMAGE_WRITE_STATIC
-#ifndef STB_IMAGE_WRITE_IMPLEMENTATION
-  #define STB_IMAGE_WRITE_IMPLEMENTATION
-#else
-  #undef STB_IMAGE_WRITE_IMPLEMENTATION
-#endif
+#define STB_IMAGE_WRITE_IMPLEMENTATION
 #include <stb_image_write.h>
-#ifndef STB_IMAGE_WRITE_IMPLEMENTATION
-  #define STB_IMAGE_WRITE_IMPLEMENTATION
-#endif
 
 namespace mlpack {
 namespace data {
 
-bool SaveImage(const std::string& filename,
-               arma::Mat<unsigned char>& image,
-               ImageInfo& info,
-               const bool fatal)
+inline bool SaveImage(const std::string& filename,
+                      arma::Mat<unsigned char>& image,
+                      ImageInfo& info,
+                      const bool fatal)
 {
   // Check to see if the file type is supported.
   if (!ImageFormatSupported(filename, true))
   {
     std::ostringstream oss;
     oss << "Save(): file type " << Extension(filename) << " not supported.\n";
     oss << "Currently image saving supports ";
-    for (auto extension : saveFileTypes)
+    for (auto extension : SaveFileTypes())
```

Comment:
I can't directly comment on the line itself, but let's switch to spaces instead of `,` to be consistent with the loading implementation, and that means we can also improve the output, so it's not `Currently image saving supports, jpg, ...` anymore.